sktime/utils/dependencies/_dependencies.py

```diff
@@ -278,6 +278,29 @@ def _check_mlflow_dependencies(msg=None, severity="error"):
 
 
 @lru_cache
+def _get_installed_packages_private():
+    """Get a dictionary of installed packages and their versions.
+
+    Same as _get_installed_packages, but internal to avoid mutating the lru_cache
+    by accident.
+    """
+    dists = distributions()
+    packages = {dist.metadata["Name"]: dist.version for dist in dists}
+    return packages
+
+
+def _get_installed_packages():
+    """Get a dictionary of installed packages and their versions.
+
+    Returns
+    -------
+    dict : dictionary of installed packages and their versions
+        keys are PEP 440 compatible package names, values are package versions
+        MAJOR.MINOR.PATCH version format is used for versions, e.g., "1.2.3"
```

Comment:
Based on the documentation, I wonder what would happen if someone installed a release candidate?

Reply:
I think version numbers are normalized to contain only numbers.
Should we remove that behaviour?

Reply:
Ok, this means it would still work? In this case, I think the proposed behavior is okay :)

Reply:
yes, a test is covering the following check: checking for versions taken from packages in the CI environments always passes. Some of these have suffixes, so if suffixes would cause problems in this way, there would be a failure, afaik (has happened before and caused the addition of normalization)

Reply:
it´s in `test_show_versions.py`